packages/eslint-plugin/tests/rules/thenable-in-promise-aggregators.test.ts

```diff
@@ -0,0 +1,561 @@
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+declare const obj: { a?: { b?: () => void } };
+await Promise.race([obj.a?.b?.()]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+declare const obj: { a: { b: { c?: () => void } } } | undefined;
+await Promise.race([obj?.a.b.c?.()]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+declare const wrappedPromise: { promise: Promise<number> };
+declare const stdPromise: Promise<number>;
+await Promise.all([wrappedPromise, stdPromise]);
+      `,
+      errors: [
+        {
+          line: 4,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+const foo = Promise;
+await foo.race([0]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+class Foo extends Promise<number> {}
+await Foo.all([0]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+const foo = (() => Promise)();
+await foo.all([0]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: 'await Promise.race?.([0]);',
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: 'await Promise?.race([0]);',
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: 'await Promise?.race?.([0]);',
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: 'await Promise.all([,]);',
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: "await Promise['all']([3]);",
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: "await Promise['race']([3]);",
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: "await Promise['allSettled']([3]);",
+      errors: [
+        {
+          line: 1,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: `
+declare const badUnion: number | string;
+await Promise.all([badUnion]);
+      `,
+      errors: [
+        {
+          line: 3,
+          messageId: 'inArray',
+        },
+      ],
+    },
+    {
+      code: 'await Promise.race(3);',
+      errors: [
+        {
+          line: 1,
+          messageId: 'nonArrayArg',
+        },
+      ],
+    },
```

Comment:
Oh, I can't understand how I didn't notice it before! `Promise.race(3)` and others with error `nonArrayArg` are not valid TS code. `all`, `race` and `allSettled` have array/iterable in their signature, so the error will be thrown by `tsc` itself, I guess we shouldn't report on this

Reply:
That's a good point! It was useful in testing locally when I was doing an eslint plugin so I could understand why I wasn't triggering the warning, but theres no reason for it to be here